pyuvdata/uvdata/uvh5.py

```diff
@@ -416,78 +631,163 @@ def _get_data(
 
         if min_frac == 1:
             # no select, read in all the data
+            inds = (np.s_[:], np.s_[:], np.s_[:], np.s_[:])
             if custom_dtype:
-                inds = (np.s_[:], np.s_[:], np.s_[:], np.s_[:])
                 self.data_array = _read_complex_astype(
                     dgrp["visdata"], inds, data_array_dtype
                 )
             else:
-                self.data_array = dgrp["visdata"][:, :, :, :]
-            self.flag_array = dgrp["flags"][:, :, :, :]
-            self.nsample_array = dgrp["nsamples"][:, :, :, :]
+                self.data_array = _index_dset(dgrp["visdata"], inds)
+            self.flag_array = _index_dset(dgrp["flags"], inds)
+            self.nsample_array = _index_dset(dgrp["nsamples"], inds)
         else:
             # do select operations on everything except data_array, flag_array
             # and nsample_array
             self._select_metadata(
                 blt_inds, freq_inds, pol_inds, history_update_string, keep_all_metadata
             )
 
+            # determine which axes can be sliced, rather than fancy indexed
+            # max_nslice_frac of 0.1 yields slice speedup over fancy index for HERA data
```

Comment:
This is a good explanatory comment. Maybe even put a reference to the PR here, because you have the profiling plots?